### PR TITLE
first version of cluster, track, track fitter, extrapolation tools and track fitter DPL workflow

### PR DESCRIPTION
**Pull request by pillot**

### CODE REVIEW

File: Detectors/MUON/MCH/Tracking/src/Cluster.cxx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+namespace o2
+{
+namespace mch
+{
+
+//_________________________________________________________________________
+Cluster::Cluster(const ClusterStruct& cl) : mX(cl.x), mY(cl.y), mZ(cl.z), mEx(cl.ex), mEy(cl.ey), mUid(cl.uid)
+{
+  /// Constructor from the simple cluster structure
+}
```
Comment (sawenzel) [inline Detectors/MUON/MCH/Tracking/src/Cluster.cxx]: this looks to me like a default copy constructor. Can be done in the header using `= default`.

Comment (pillot) [reply Detectors/MUON/MCH/Tracking/src/Cluster.cxx]: The copy constructor is already = default in the header. This one is a constructor of an internal Cluster object from a simple clusterStruct.

Comment (sawenzel) [reply Detectors/MUON/MCH/Tracking/src/Cluster.cxx]: Ok didn't notice this.

File: Detectors/MUON/MCH/Tracking/src/TrackExtrap.cxx [hunk 2]
```diff
@@ -0,0 +1,716 @@
...
+    return !tooManyStep;
+  }
+}
+
+//__________________________________________________________________________
+bool TrackExtrap::extrapOneStepRungekutta(double charge, double step, const double* vect, double* vout)
+{
+  /// <pre>
+  ///  ******************************************************************
+  ///  *                                                                *
```
Comment (sawenzel) [inline Detectors/MUON/MCH/Tracking/src/TrackExtrap.cxx]: out of curiosity: Do we really need to code another solver for MUON tracks? Isn't this available somewhere generically? @shahor02 ?

Comment (shahor02) [reply Detectors/MUON/MCH/Tracking/src/TrackExtrap.cxx]: @sawenzel No, Muon has to use completely different track model, for the propagation in dipole field...

Comment (sawenzel) [reply Detectors/MUON/MCH/Tracking/src/TrackExtrap.cxx]: Thanks.

**Pull request merged**